hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateDagVisitor.java

```diff
@@ -186,6 +187,32 @@ public Vertex onMapIndexScan(IndexScanMapPhysicalRel rel) {
                 );
     }
 
+    public Vertex onCalc(CalcPhysicalRel rel) {
+        // TODO [sasha] : program might be trivial [program.isTrivial()], we may just skip vertex creation
+        RexProgram program = rel.getProgram();
+        List<Expression<?>> projection = rel.projection(parameterMetadata);
+
+        Vertex project = dag.newUniqueVertex("Project", mapUsingServiceP(
+                ServiceFactories.nonSharedService(ctx ->
+                        ExpressionUtil.projectionFn(projection, ExpressionEvalContext.from(ctx))),
+                (Function<JetSqlRow, JetSqlRow> projectionFn, JetSqlRow row) -> projectionFn.apply(row)
+        ));
+
+        if (program.getCondition() != null) {
+            Expression<Boolean> filterExpr = rel.filter(parameterMetadata);
+
+            Vertex filter = dag.newUniqueVertex("Filter", filterUsingServiceP(
+                    ServiceFactories.nonSharedService(ctx ->
+                            ExpressionUtil.filterFn(filterExpr, ExpressionEvalContext.from(ctx))),
+                    (Predicate<JetSqlRow> filterFn, JetSqlRow row) -> filterFn.test(row)));
+            dag.edge(between(filter, project));
+            connectInputPreserveCollation(rel, filter);
+        } else {
+            connectInputPreserveCollation(rel, project);
+        }
```

Comment:
We don't have to create two vertices here. The `map` transform can be used for both mapping and filtering - if it returns `null`.